core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveCodec.java

```diff
@@ -260,4 +265,65 @@ protected String debugInfo() {
     return this.toString();
   }
 
+  public static VectorUtil checkAndUpdateToChildVector(ColumnVectorInfo vectorInfo, int pageSize,
+      CarbonColumnVector vector, DataType vectorDataType) {
+    VectorUtil vectorUtil = new VectorUtil(pageSize, vector, vectorDataType);
+    Stack<CarbonColumnVector> vectorStack = vectorInfo.getVectorStack();
+    // check and update to child vector info
+    if (vectorStack != null && vectorStack.peek() != null && vectorDataType.isComplexType()) {
+      if (DataTypes.isArrayType(vectorDataType)) {
+        List<Integer> childElementsCountForEachRow =
+            ((CarbonColumnVectorImpl) vector.getColumnVector())
+                .getNumberOfChildrenElementsInEachRow();
+        int newPageSize = 0;
+        for (int childElementsCount : childElementsCountForEachRow) {
+          newPageSize += childElementsCount;
+        }
+        vectorUtil.setPageSize(newPageSize);
+      }
+      // child vector flow, so fill the child vector
+      CarbonColumnVector childVector = vectorStack.pop();
+      vectorUtil.setVector(childVector);
+      vectorUtil.setVectorDataType(childVector.getType());
+    }
+    return vectorUtil;
+  }
+
+  // Utility class to update current vector to child vector in case of complex type handling
+  public static class VectorUtil {
+    private int pageSize;
+    private CarbonColumnVector vector;
+    private DataType vectorDataType;
+
+    private VectorUtil(int pageSize, CarbonColumnVector vector, DataType vectorDataType) {
+      this.pageSize = pageSize;
+      this.vector = vector;
+      this.vectorDataType = vectorDataType;
+    }
+
+    public int getPageSize() {
```

Comment:
we can use org.apache.commons.lang3.tuple class for returning 3 args from one method , pls check if you can use here

Reply:
removed this whole class itself and updating vector inside **ColumnarVectorWrapperDirectFactory
          .getDirectVectorWrapperFactory**